node/core/approval-voting/src/lib.rs

```diff
@@ -689,28 +765,49 @@ async fn run<C>(
 
 				actions
 			}
-			background_request = background_rx.next().fuse() => {
-				if let Some(req) = background_request {
-					handle_background_request(
-						&mut ctx,
-						&mut state,
-						&subsystem.metrics,
-						req,
-					).await?
-				} else {
-					Vec::new()
+			approval_state = currently_checking_set.next().fuse() => {
+				let mut actions = Vec::new();
+				let sender = ctx.sender();
+				if let Some((
+					relay_block_hashes,
+					ApprovalState {
+						validator_index,
+						candidate_hash,
+						candidate_index,
+						status
+					}
+				)) = approval_state {
+					if matches!(status, ApprovalOutcome::Approved) {
+						for block_hash in relay_block_hashes.into_iter() {
+							if let Ok(mut _actions) = issue_approval(
+								&mut sender.clone(),
+								&mut state,
+								&subsystem.metrics,
+								ApprovalVoteRequest {
+									validator_index,
+									block_hash,
+									candidate_index,
+								}
+							) {
+								actions.append(&mut _actions);
+							}
+						}
+					}
+					currently_checking_set.approvals_cache.put(candidate_hash, status);
+					let _ = currently_checking_set.candidate_hash_map.remove(&candidate_hash);
```

Comment:
This is either redundant or something `current_checking_set` should take care of internally.